### ISSUE TITLE
First redirect on router "hash" mode

### ISSUE THREAD

**Issue by SnirShechter** [nuxt/nuxt.js#5241]
### Version

[2.4.0](https://github.com/nuxt.js/releases/tag/2.4.0)

### Reproduction link

[https://codesandbox.io/s/z31mzrqjlp](https://codesandbox.io/s/z31mzrqjlp)

### Steps to reproduce

*Existing CodeSandbox:*
1. enter the URL
2. index.vue page will attempt to load, the middleware will redirect to {name:'secret'} and fail (wrong URL)

*New Project:*
1. create a project with config:
{
  mode: 'spa',
  router:{
    mode:'hash'
  }
}
2. add a middleware with a redirect ( redirect({name:routeName}); )
3. generate the app.
4. redirect will fail with an extra 

### What is expected ?

To redirect fine

### What is actually happening?

An extra '/' is added, and the redirect fails
![Untitled.png](https://cmty.s3.eu-west-1.amazonaws.com/71995937/88aad6a0-4656-11e9-9e61-d12619e82669.png)

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8834">#c8834</a>)</em></sub></div>

**Comment by SnirShechter** [comment]
UPDATE: The issue is probably in formatUrl function [here](https://github.com/nuxt/nuxt.js/blob/5f9b1044ff2eaadb64e3c172b023616c2fb1f512/packages/vue-app/template/utils.js#L555).

Can anyone please check? it fails on hash mode.

**Comment by SnirShechter** [comment]
Seems like this is solved in **2.5.1**

**Comment by dyzajash** [comment]
Error occured again in version 2.6.3.

```
export default function({ app, store, redirect }) {
  if (! store.state.auth.isAuthenticated) {
    redirect({ name: 'auth-login' })
  }
}
```


`Failed to execute 'replace' on 'Location': '//#/auth/login' is not a valid URL.`
